pojo/src/main/java/com/connexta/transformation/pojo/unknown/UnknownMetadataPojo.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.transformation.pojo.unknown;
+
+import com.connexta.transformation.pojo.MetadataPojo;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+
+/**
+ * This class provides a pojo implementation for unknown request information objects and is used for
+ * forward compatibility where the current code might not be able to understand a new type of
+ * resource information. In such case, Jackson would map this new type of request information object
+ * to this class.
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(Include.NON_NULL)
+@JsonTypeName("unknown")
+public class UnknownMetadataPojo extends MetadataPojo implements UnknownPojo {}
```

Comment:
❓ When would these unknown pojos be used? Wouldn't an object with unknown fields just be read into the usual pojo which would then be flagged to contain unknowns?

Reply:
This is to allow subclassing in the future while still handling it properly. So additional fields are ignored and old ones are converted and new classes are read by default but yet reported as unknown.